src/Model/Scope/AbstractScope.php

```diff
@@ -0,0 +1,225 @@
+     *
+     * @return bool
+     */
+    abstract public function toWords(bool $asHtml = false): string;
+
+    /**
+     * Simplifies by peeling off nested scopes with single contained component.
+     * Useful for converting (((field = value))) to field = value.
+     *
+     * @return AbstractScope
+     */
+    public function simplify()
+    {
+        return $this;
+    }
+
+    /**
+     * Sets the scope as excluded from applying it to the model.
+     *
+     * @return $this
+     */
+    public function deactivate()
+    {
+        $this->active = false;
+
+        return $this;
+    }
+
+    /**
+     * Sets the scope as included for applying it to the model.
+     *
+     * @return $this
+     */
+    public function activate()
+    {
+        $this->active = true;
+
+        return $this;
+    }
+
+    /**
+     * Returns is scope should be applied to the model.
+     */
+    public function isActive(): bool
+    {
+        return $this->active && !$this->isEmpty();
+    }
+
+    /**
+     * Returns if scope contains several conditions.
+     */
+    public function isCompound(): bool
+    {
+        return false;
+    }
+
+    /**
+     * Register object specific placeholoder for a value to be replaced
+     * The $options array may contain
+     * - label : string - the label to use when converting toWords
+     * - value : string|Callable - the actual value to be used when applying the scope
+     * If value is Callable the it is called with $model, $scope as arguments.
+     *
+     * @param string|callable $options
+     */
+    public function registerPlaceholder(string $key, $options)
+    {
+        $this->placeholders[$key] = is_array($options) ? $options : [
+            'label' => is_string($options) ? $options : $key,
+            'value' => $options,
+        ];
+
+        return $this;
+    }
+
+    /**
+     * Register several placeholders from an array.
+     *
+     * @see AbstractScope::registerPlaceholder
+     */
+    final public function registerPlaceholders(array $placeholders)
+    {
+        foreach ($placeholders as $key => $options) {
+            $this->registerPlaceholder($key, $options);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Register global placeholoder for a value to be replaced
+     * The $options array may contain
+     * - label : string - the label to use when converting toWords
+     * - value : string|Callable - the actual value to be used when applying the scope
+     * If value is Callable the it is called with $model, $scope as arguments.
+     *
+     * @param string|callable $options
+     */
+    final public static function registerGlobalPlaceholder(string $key, $options)
+    {
+        self::$globalPlaceholders[$key] = is_array($options) ? $options : [
+            'label' => is_string($options) ? $options : $key,
+            'value' => $options,
+        ];
+    }
+
+    public static function __set_state($array)
```

Comment:
usecase?

Reply:
Method is called when using `var_export($scope)`.
I use this when creating a script or commant to be executed in the background on a separate process not dependent on current PHP process.